clean_cache/action.yml

```diff
@@ -0,0 +1,90 @@
+name: 'clean_cache'
+description: 'clean the cache'
+inputs:
+  s3cmd:
+    type: string
+    required: true
+    description: 'command line tool'
+  cache_endpoint:
+    type: string
+    required: true
+    description: 'secrets.CACHE_ENDPOINT for actions-cache'
+  cache_access_key:
+    type: string
+    required: true
+    description: 'secrets.CACHE_ACCESS_KEY for actions-cache'
+  cache_secret_key:
+    type: string
+    required: true
+    description: 'secrets.CACHE_SECRET_KEY  for actions-cache'
+  cache_objects:
+    type: string
+    required: true
+    description: 'objects to delete for cleanup'
+
+runs:
+  using: composite
+  steps:
+    - name: Install s3cmd and gh
+      if: ${{ inputs.install_ghcli == 'true' }}
+      shell: bash
+      run: |
+        GH=$(type -p 'gh'||true)
+        if [ -z "${GH}" ]; then
+          echo -e "\033[0;35m###  Installing gh ###\033[0m"
+          curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg
+          sudo chmod go+r /usr/share/keyrings/githubcli-archive-keyring.gpg
+          ARCH=$(dpkg --print-architecture)
+          echo "deb [arch=${ARCH} signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" |\
+          sudo tee /etc/apt/sources.list.d/github-cli.list >/dev/null
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -qq update
+          sudo apt-get -qq install gh
+          GH=$(type -p 'gh'||true)
+        fi
+        ${GH} extension install actions/gh-actions-cache
+        sudo DEBIAN_FRONTEND=noninteractive apt-get -qq install s3cmd
+        RUNS_ON='${{ inputs.runs_on }}'
+        if [[ "${RUNS_ON}" != *"ubuntu" ]]; then
+          cat >/home/runner/.s3cfg <<EOF
+        [default]
+        check_ssl_certificate = False
+        check_ssl_hostname = False
+        host_base = ${{inputs.cache_endpoint}}
+        host_bucket = %(bucket)s.${{inputs.cache_endpoint}}
+        accessKey: ${{ inputs.cache_access_key }}
+        secretKey: ${{ inputs.cache_secret_key }}
+        EOF
+          chmod 0600 /home/runner/.s3cfg
+        fi
+
+    - name: Cleanup cached github objects
+      if: ${{ steps.generate_report.outputs.overall_status == 'success' }}
+      shell: bash
+      run: |
+        GH=$(type -p 'gh'||true)
+        if [ -n "${GH}" ]; then
+         # yamllint disable-line rule:line-length
+           for CACHE in ${{ inputs.cache_objects }} ; do
+            ID_LIST="$(${GH} actions-cache list --key ${CACHE}|awk '{print $1}'||true)"
+              for ID in ${ID_LIST}; do
+              echo -e "\033[0;35m###  Cleaning cached object '${ID}' from github ###\033[0m"
+                ${GH} actions-cache delete "${ID}" --confirm
+              done
+            done
+          fi
+
+    - name: Clear s3 cache
+      if: ${{ inputs.runs_on != '"ubuntu-latest"' }}
```

Comment:
Where is inputs.runs_on defined?

Reply:
it is defined in universal workflow but now used in clear_cache too